File: source/frontend/src/modules/groups/types/group.request.ts
import { GroupMemberRole } from './group.enum'

export interface CreateGroupRequest {
  name: string
  subjectOrProjectName: string
  limitedUser: number
}

export interface UpdateGroupRequest {
  name?: string
  description?: string
  subjectOrProjectName?: string
}

export interface AddMemberRequest {
  userId?: number
  role?: GroupMemberRole
  userName?: string
  email?: string
  avatarUrl?: string
  userCode?: string
  isActive?: boolean
  userRole?: string
}

export interface UpdateMemberRoleRequest {
  userId: number
  newRole: GroupMemberRole
}

export interface RemoveMemberRequest {
  userId: number
}

File: source/backend/project.Presentation/Models/GroupRequest.cs
using project.Domain.Models;

namespace project.Presentation.Models
{
    public class CreateGroupRequest
    {
        public string Name { get; set; } = string.Empty;
        public string SubjectOrProjectName { get; set; } = string.Empty;
        public int LimitedUser { get; set; }
    }
    public class UpdateGroupRequest
    {
        public string Name { get; set; }
        public string SubjectOrProjectName { get; set; }
    }

    public class AddMemberRequest
    {
        public int UserId { get; set; }
        public GroupMemberRole Role { get; set; } = GroupMemberRole.Member;
    }

    public class UpdateMemberRoleRequest
    {
        public GroupMemberRole NewRole { get; set; }
    }
    public class RepoUpdateRequest
    {
        public string RepoUrl { get; set; }
    }

}
